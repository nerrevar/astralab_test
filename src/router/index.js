import Vue from 'vue'
import VueRouter from 'vue-router'

import EnterView from '@/views/EnterView.vue'
import SignUpView from '@/views/SignUpView.vue'
import SignInView from '@/views/SignInView.vue'

Vue.use(VueRouter)

const routes = [
  {
    path: '/',
    name: 'EnterView',
    component: EnterView,
  },
  {
    path: '/sign_up',
    name: 'SignUpView',
    component: SignUpView,
  },
  {
    path: '/sign_in',
    name: 'SignInView',
    component: SignInView,
  },
  {
    path: '*',
    redirect: '/',
  }
]

const router = new VueRouter({
  mode: 'history',
  base: process.env.BASE_URL,
  routes,
})

export default router
