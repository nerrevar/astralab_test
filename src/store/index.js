import Vue from 'vue'
import Vuex from 'vuex'

import firebase from 'firebase/app'
import 'firebase/auth'

Vue.use(Vuex)

const firebaseConfig = {
  apiKey: 'AIzaSyC4VT28fW5urLc3iRZmidKIPRF_Wkha7rw',
  authDomain: 'astralabtest.firebaseapp.com',
  projectId: 'astralabtest',
  storageBucket: 'astralabtest.appspot.com',
  messagingSenderId: '836632717701',
  appId: '1:836632717701:web:d11044d113209431cf7cec',
}

firebase.initializeApp(firebaseConfig)

export default new Vuex.Store({
  state: {
    firebaseApp: firebase.app(),
    user: { displayName: 'John Doe' },
  },
  getters: { getFirebaseApp: state => state.firebaseApp },
  mutations: { setUser: (state, payload) => state.user = payload.user },
  actions: {},
  modules: {},
})
