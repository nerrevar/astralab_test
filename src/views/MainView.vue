<template>
  <div class="wrapper">
    <div class="title">
      Hello, {{ user.name }}!
    </div>
    <img
      class="main-img"
      src="@/assets/woman_main.svg"
      alt=""
    />
  </div>
</template>

<script>
import { mapState } from 'vuex'

export default {
  name: 'MainView',
  computed: mapState({ user: state => state.user }),
}
</script>

<style lang="sass" scoped>
.wrapper
  justify-content: space-between

.title
  margin-top: 40%
</style>
