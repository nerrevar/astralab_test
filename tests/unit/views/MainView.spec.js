import { shallowMount, createLocalVue } from '@vue/test-utils'
import VueRouter from 'vue-router'
import Vuex from 'vuex'

import MainView from '@/views/MainView.vue'

import router from '@/router'
import store from '@/store/'

const localVue = createLocalVue()

localVue.use(VueRouter)
localVue.use(Vuex)

describe('MainView.vue', () => {
  const vm = shallowMount(
    MainView,
    {
      router,
      store,
    }
  )

  it('renders', () => {
    expect(vm.element).toMatchSnapshot()
  })

  it('title text', () => {
    expect(vm.find('.title').text()).toBe('Hello, John Doe!')
  })
})
