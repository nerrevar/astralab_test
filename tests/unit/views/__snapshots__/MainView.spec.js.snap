// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainView.vue renders 1`] = `
<div
  class="wrapper"
>
  <div
    class="title"
  >
    
    Hello, John Doe!
  
  </div>
   
  <img
    alt=""
    class="main-img"
    src="@/assets/woman_main.svg"
  />
</div>
`;
